### PR TITLE
FIX#4259: Add better error handling for getting AWS credentials for a challenge 

### PR DESCRIPTION
**Pull request by AishwaryJain07**
Here, I added Try-catch block to look for the errors, and also added the looging message to indicate the retrieval of challenge_phase id and passes a m,message after it retrieves of challenge_phase id.

### CODE REVIEW

**Comment by AishwaryJain07** [comment]
closes #4259

**Comment by AishwaryJain07** [comment]
@Suryansh5545 @gchhablani @RishabhJain2018 can you please review

File: apps/challenges/views.py [hunk 1]
```diff
@@ -2830,24 +2830,33 @@ def get_aws_credentials_for_participant_team(request, phase_pk):
...
-    participant_team = get_participant_team_of_user_for_a_challenge(
-        request.user, challenge.pk
-    )
-    if not challenge.is_docker_based:
-        response_data = {
-            "error": "Sorry, this is not a docker based challenge."
-        }
-        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+    try:
+        logging.debug(f"Retrieving challenge phase for ID: {phase_pk}")
```
Comment (Suryansh5545) [inline apps/challenges/views.py:2834]: Could you please provide an explanation for why this change was added?

File: apps/challenges/views.py [hunk 2]
```diff
@@ -2830,24 +2830,33 @@ def get_aws_credentials_for_participant_team(request, phase_pk):
...
+        challenge_phase = get_challenge_phase_model(phase_pk)
+        logging.debug(f"Challenge phase retrieved: {challenge_phase}")
```
Comment (Suryansh5545) [inline apps/challenges/views.py:2836]: Same here @AishwaryJain07

**Review by Suryansh5545 — COMMENTED** [review]
Left some comments to be addressed

File: apps/challenges/views.py (hunk 1 above)
Comment (AishwaryJain07) [reply apps/challenges/views.py:2834]: @Suryansh5545 it creates a log entry, developers can access for troubleshooting and understanding the code's behavior.

the idea of logging was mentioned by @gchhablani in his pr, "https://github.com/Cloud-CV/EvalAI/pull/4261".

this was all the behaviour i tried to add in the code, kindly let me know further any.
plus this also adds the try-catch block which already @gchhablani did.

**Comment by AishwaryJain07** [comment]
> Left some comments to be addressed

@Suryansh5545 the log only creates an entry for developers to understand.
like a debugging message.

File: apps/challenges/views.py (hunk 1 above)
Comment (Suryansh5545) [reply apps/challenges/views.py:2834]: @AishwaryJain07 i think you misunderstood it. we want to give a error message to the user as return if something goes wrong instead of a generic message. you are going in the right direction by capturing the error and sending it back as a response. but the log thing isn't required for now.

Comment (gchhablani) [reply apps/challenges/views.py:2834]: @AishwaryJain07 We do not want logging.
